### DISCUSSION TITLE
Resizing in both directions

### DISCUSSION THREAD

**Discussion by chojecik** [react-grid-layout/react-grid-layout#1647] [Q&A]
Hi guys,

I'd like my element to increase/decrease in the same way on x and y axis while resizing. So basically I need to preserve the original shape of an element and just scale it while resizing. Does anyone of you knows how to achieve that? I've been trying to use onResize callback and manually add values to x/y but I didn't make it

**Comment by alban44980** [comment]
Hi ! I'd also be down to know if anyone managed to do this :)

**Comment by chojecik** [comment]
It was not pretty and finally I haven't even used it due to changed requirements but I was able to achieve that using onResize callback after all. You just need to add or substract values from each axis when onResize is called which is on every single pixel's change. Bad news is that it won't change dynamically while resizing only after you release the mouse button so it looks kinda weird.

**Comment by alban44980** [comment]
Cheers! Just managed to make it work too ! You can actually make it happen dynamically, taken from the example here: https://github.com/react-grid-layout/react-grid-layout/blob/master/test/examples/10-dynamic-min-max-wh.jsx .
